RFCs/005-Framework-Extensibility-Custom-DataSource.md

```diff
@@ -1,58 +1,81 @@
-# RFC 005- Framework Extensibility for Custom DataSource
+# RFC 005- Framework Extensibility for Custom Test Data Source
 
 ## Summary
 This details the MSTest V2 framework extensibility for specifying custom data source for data driven tests.
 
 ## Motivation
-Often times, custom data source is required for data driven tests. User should be able to leverage test framework extensibility to provide custom data sources for test execution.
+Often times, custom data sources are required for data driven tests. User should be able to leverage test framework extensibility to provide custom data sources for test execution.
 
 ## Detailed Design
 
 ### Requirements
-1. Test data from custom data source should be easily associated with multiple test cases.
+1. A cusotm data source can be used by multiple test cases. 
+2. A test case can have multiple data sources.
 
 ### Proposed solution
-Here is a solution using which test data can be easily assocaited with test case.
+Here is a solution for using custom data source in data driven tests.
 
-The test framework should define an interface `IDataSource` which can be extended to get data from custom data source. The defintion of this interface is as below :
+The test framework should define an abstract class `TestDataSource` which can be extended to get data from custom data source.
 ```
-    public interface IDataSource
+    public abstract TestDataSource : Attribute
```

Comment:
@harshjain2 I think you could probably get that done in one line of code by doing something like this

```
ITestDataSource[] dataSources = testMethod.GetAllAttributes(true).Where(a => a is ITestDataSource).OfType<ITestDataSource>().ToArray();
```
Then your code would works pretty the same as it is now. 

Making it an attribute is saying "No developer will ever want to make a data source attribute that inherits from anything other than TestDataSourceAttribute". Is this an assumption that is safe to make?